.chloggen/kafka-clientconfig-tls-kafkaexporter.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: kafkaexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecate `auth::tls` and introduce `tls` config
```

Comment:
The `auth.tls` is already using `*configtls.ClientConfig`. 

Then why do we need to introduce this breaking change? 

The https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37776 could be resolved by a doc change.

Reply:
Sorry, I should have included this in the PR description. Two reasons:
 - TLS is primarily used for encryption. It might also be for client authentication, but that's less common.
 - Consistency with the majority of other components using configtls.ClientConfig. Almost all of them put it at the top-level.

Put together, I think this makes for a clearer structure. Having consistency makes it easier to understand the configuration, particularly if you're already familiar with other components. Moving out of "auth" will also make it easier to understand if you're not using TLS for auth.